### PR TITLE
Replace deprecated code

### PR DESCRIPTION
**Pull request by smpark7**
## Description
This PR replaces deprecated code and input file parameters. All test results remain unchanged.
This PR seems big in terms of lines edited because of changes to the MOOSE input file format (bullet points 2, 3, 5, 6). But all changes are essentially quite basic changes.

## Changes
- Replace `ModulesApp::registerAll` with `ModulesApp::registerAllObjects<MoltresApp>`
- Remove `./` and `../` from headers and closers in input files in accordance with the latest HIT input file structure
- Replace `f_name`, `function`, and `args` parameters in `DerivativeParsedMaterial` objects with `property_name`, `expression`, and `coupled_variables`
- Add whitespaces as necessary around `=` signs and in commented lines
- Replace `value` with `expression` in `Function` objects
- Move `coord_type` from `[Problem]` to `[Mesh]`
- Replace `DelayedNeutronEigenSource` with `DelayedNeutronSource` in `tutorial/eigenvalue/nts.i` and in the tutorial docs because of #237
- Update Squirrel submodule for the latest commits in that repo involving similar deprecation changes

## Impact
- Brings Moltres up-to-date with current MOOSE standards
- Fixes tutorial input file which would provide an error otherwise

### CODE REVIEW

File: include/postprocessors/ElmIntegTotFissNtsPostprocessor.h [hunk 1]
```diff
@@ -14,8 +14,26 @@ class ElmIntegTotFissNtsPostprocessor : public ElementIntegralPostprocessor
...
 
+  // The number of neutron energy groups.
   int _num_groups;
+
+  // The number of precursor groups.
+  int _num_precursor_groups;
+
+  // Whether to account for delayed neutrons
+  bool _account_delayed;
+
```
Comment (yardasol) [inline include/postprocessors/ElmIntegTotFissNtsPostprocessor.h]: Helpful comments!

File: problems/2021-cnrs-benchmark/phase-0/nts.i [hunk 2]
```diff
@@ -37,7 +37,6 @@
   vacuum_boundaries = 'bottom left right top'
   create_temperature_var = false
   eigen = true
-  transient = false
```
Comment (yardasol) [inline problems/2021-cnrs-benchmark/phase-0/nts.i]: Why is this word removed? Wouldn't it change the behavior? Or is it now deprecated?

File: problems/2021-cnrs-benchmark/phase-2/README.MD [hunk 3]
```diff
@@ -6,13 +6,14 @@ Blue Waters supercomputer with 512 MPI processes on 32 XK or 16
...
 transient case with the shortest perturbation period
-(`transient-perturb-7.i`) took ~24h to complete
-5000 timesteps and reach reasonable convergence of the gain and phase shift
-values on the 512 MPI processes on Blue Water. `transient-perturb-1.i` took
-~125h to complete 4000 timesteps under the same conditions.
+took ~0.67h to complete one period (200 timesteps) and 16.67h in total (5000
+timesteps) to reach reasonable convergence of the gain and phase shift
+values on the 512 MPI processes on Blue Water. Conversely, the case with the
+longest perturbation period took x2.25 longer due to the longer simulation time
+needed to reach convergence.
```
Comment (yardasol) [inline problems/2021-cnrs-benchmark/phase-2/README.MD]: Is this update from the most recent paper, [2]?

File: problems/2021-cnrs-benchmark/phase-2/README.MD [hunk 4]
```diff
@@ -6,13 +6,14 @@ Blue Waters supercomputer with 512 MPI processes on 32 XK or 16
 XE computing nodes. We recommend running these time dependent simulations on
 clusters with at least 64 independent processors and 64 GB of RAM. The
 transient case with the shortest perturbation period
-(`transient-perturb-7.i`) took ~24h to complete
-5000 timesteps and reach reasonable convergence of the gain and phase shift
-values on the 512 MPI processes on Blue Water. `transient-perturb-1.i` took
-~125h to complete 4000 timesteps under the same conditions.
+took ~0.67h to complete one period (200 timesteps) and 16.67h in total (5000
+timesteps) to reach reasonable convergence of the gain and phase shift
+values on the 512 MPI processes on Blue Water. Conversely, the case with the
```
Comment (yardasol) [inline problems/2021-cnrs-benchmark/phase-2/README.MD]:
```suggestion
values on the 512 MPI processes on Blue Waters. Conversely, the case with the
```

**Review by yardasol — APPROVED** [review]
Hi @smpark7, good changes here. Just a few small questions and one typo fix, then I think this is ready to merge!

File: problems/2021-cnrs-benchmark/phase-2/README.MD (hunk 3 above)
Comment (smpark7) [reply problems/2021-cnrs-benchmark/phase-2/README.MD]: Yea, the old compute times were based on preliminary results.

File: problems/2021-cnrs-benchmark/phase-0/nts.i (hunk 2 above)
Comment (smpark7) [reply problems/2021-cnrs-benchmark/phase-0/nts.i]: This change is actually part of #237 which we just merged. This object never required `transient` as an input parameter. But MOOSE tests are now stricter with unused parameters in the input file.

**Comment by smpark7** [comment]
@yardasol Thanks for your review! I fixed the typo and replied to your questions. Let me know if you have any other comments.

**Review by yardasol — APPROVED** [review]
Good job @smpark7

**Pull request merged**